src/evaluate/evaluator.py

```diff
@@ -12,95 +12,125 @@
             for element in predictions
         ]
 
-    def get_references(self):
-        return self.data["references"]
-
-    def statistic(self, predictions=None, references=None, metric_key: Optional[str] = None):
-        return self.metric.compute(predictions=predictions, references=references)[
-            self.metric_key if metric_key is None else metric_key
-        ]
-
-    def get_confidence_interval(self, confidence_level: float = 0.95, n_resamples: int = 9999):
-        predictions = self.get_predictions()
-        references = self.get_references()
-        return bootstrap(data=(predictions, references), statistic=self.statistic, paired=True, vectorized=False)
-
-    def get_bootstrap_p_value(
+    def compute(
         self,
-        baseline_pipe: Union[str, Pipeline],
-        baseline_label_mapping: Optional[Dict[str, Number]] = None,
+        model: str = None,
+        pipe: Union[Pipeline, Callable] = None,
+        data: Union[str, Dataset] = None,
+        metric: Union[str, Metric] = None,
+        strategy: Literal["simple", "bootstrap"] = "simple",
+        confidence_level: float = 0.95,
         n_resamples: int = 9999,
-    ):
-        """Compare two model's scores by getting p-values from non-parametric bootstrap. Here, the null hypothesis is:
-        "current model is no better than the baseline model on the test population".
-        The implementation is based on the paper: https://aclanthology.org/D12-1091.pdf.
-        """
-        baseline_pipe = pipeline(baseline_pipe) if isinstance(baseline_pipe, str) else baseline_pipe
-        preds = np.array(self.get_predictions())
-        baseline_preds = np.array(
-            self.get_predictions(baseline_pipe=baseline_pipe, baseline_label_mapping=baseline_label_mapping)
+        input_column: str = "inputs",
+        label_column: str = "references",
+        label_mapping: Optional[Dict[str, Number]] = None,
+    ) -> Tuple[Dict[str, float], Any]:
+        assert data is not None
+
+        pipe = (
+            pipeline(self.task, model=model)
+            if model is not None
+            else pipe
+            if pipe is not None
+            else pipeline(self.task)
         )
-        references = np.array(self.get_references())
-
-        delta = self.statistic(preds, references) - self.statistic(baseline_preds, references)
-        s = 0
-        for i in range(n_resamples):
-            resampled_indices = np.random.choice(len(references), size=len(references), replace=True)
-            delta_i = self.statistic(preds[resampled_indices], references[resampled_indices]) - self.statistic(
-                baseline_preds[resampled_indices], references[resampled_indices]
+        assert pipe.task == self.task
+
+        metric = (
+            load_metric(metric)
+            if isinstance(metric, str)
+            else (metric if metric is not None else (self.default_metric if self.default_metric is not None else None))
```

Comment:
`self.default_metric` is also a string, right? This would fail in that case as `metric` would end up as the default string.